core/src/main/scala/org/apache/spark/util/UninterruptibleThread.scala

```diff
@@ -100,8 +100,12 @@ private[spark] class UninterruptibleThread(
      * @return true when there is no concurrent [[runUninterruptibly()]] call ([[uninterruptible]]
      *         is true) and no concurrent [[interrupt()]] call, otherwise false
      */
-    def isInterruptible: Boolean = synchronized {
-      shouldInterruptThread = uninterruptible
+    def isInterruptible(t: Thread): Boolean = synchronized {
```

Comment:
This is in an inner class of `UninterruptibleThread`, can we access the outer `this` directly?